### ISSUE TITLE
Automatically always copy screenshot to clipboard?

### ISSUE THREAD

**Issue by orschiro** [mate-desktop/mate-utils#177]
I am wondering whether we can save people clicks if screenshots are always automatically copied to clipboard?

Is there any downside to automatically storing each taken screenshot directly in the clipboard?

In this case, the dialogue window would only need the two buttons "Close" and "Save".

Would that be more elegant?

**Comment by shevegen** [comment]
Perhaps an option to a widget could be used for that. By default, it is not clicked on. If set to on, then the above behaviour could be employed.

I am not a UI expert; my only exposure I have from the code side to GTK was via the ruby bindings, but perhaps a tab could be shown and people could select that tab for "extended options", where the above could reside.

**Comment by MarioMey** [comment]
I think this option should be at command level. I screenshot areas a lot. I should create a shortcut Alt-PrtScreen with:

`mate-screenshot -a -c`

Where -a is for area, -c is for copying directly to clipboard, with no dialog after capturing.

```
$ mate-screenshot --help
Use:
  mate-screenshot [OPCIÓN…] Capture screenshot
...
Aplication options:
...
-c, --copy      Capture directly to clipboard, no dialog after capturing.
...
```

**Comment by adefaria** [comment]
Yeah like old gnome-screenshot used to offer. Truth is mate-screenshot does offer this but just doesn't document it.

**Comment by MarioMey** [comment]
Well, it is part of mate-screenshot since dontknow.

```
mariomey@debian11:~$ mate-screenshot --help
Uso:
  mate-screenshot [OPCIÓN…] Capturar imagen de la pantalla

Opciones de ayuda:
  -h, --help                     Mostrar opciones de ayuda
  --help-all                     Muestra todas las opciones de ayuda
  --help-gtk                     Mostrar opciones GTK+

Opciones de la aplicación:
  -w, --window                   Capturar una ventana en vez de la pantalla entera
  -a, --area                     Capturar un área de la pantalla en vez de la pantalla entera
  -c, --clipboard                Enviar área capturada directamente al portapapeles
  -b, --include-border           Incluir el borde de la ventana en la captura de pantalla
  -B, --remove-border            Eliminar el borde de la ventana de la captura de pantalla
  -d, --delay=segundos           Tomar una captura de pantalla después del retraso especificado [en segundos]
  -e, --border-effect=efecto     Efecto para añadir al borde (sombra, borde o ninguno)
  -i, --interactive              Establecer las opciones interactivamente
  --version                      Imprimir información de la versión y salir
  --display=VISOR                Visor [display] X que usar

```

**Comment by adefaria** [comment]
Sorry, I was not clear. mate-screenshot lacks the -f option. I often want simply to take a screenshot of an area of my screen and say send it to somebody else. I bind the "Print" key to do a capture of the screen by area. Since some places accept the clipboard for pasting images (e.g. thunderbird) others do not (e.g. voice.google.com), I had a little script to call gnome-screenshot and compose a file name (Screenshot-<date>-<time>) to a specific directory so I could either paste or find the screenshot image and drag it into where I wanted to post it. Also, since they were all in the same directory it was easier for me to clean up unneeded screenshots later.

When I moved to Mate I tried using mate-screenshot but you can't screenshot into a file via the command line option. You can if you use the GUI or interactive mode (-i). For the life of me, I cannot fathom why they took that option out! It wasn't hurting anybody...

**Comment by MarioMey** [comment]
> screenshot into a file via the command line option

You can try something like:
```bash
#!/bin/bash
notify-send 'Automatic Screenshot to file' -t 600
sleep 0.1 # This should / could fix an issue in mate-screenshot, try to remove it and maybe it is not necesary.
mate-screenshot -a -c
xclip -selection clipboard -t image/png -o > ~/algo.png
```

This would create file "algo.png" in home dir. You should manage multiple file creations (with numbers or something).

**Comment by adefaria** [comment]
Well, I had just gone back to gnome-screenshot but this looks like it'll work. Thanks!

Still seems odd to remove functionality that ain't broke, is useful and its removal takes away functionality.
